### ISSUE TITLE
Cannot read property 'toString' of null

### ISSUE THREAD

**Issue by khaledosman** [Azure-Samples/azure-sdk-for-js-storage-blob-upload-download#4]
### This issue is for a: (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> clone the repo and run `node index.js`

### Any log messages given by the failure
 ```
Container: "demo" is created
Blob "quickstart.txt" is uploaded
Local file "./readme.md" is uploaded
Local file "./readme.md" is uploaded as a stream
Blobs in "demo" container:
 - quickstart.txt
 - readme-stream.md
 - readme.md
TypeError: Cannot read property 'toString' of null
    at execute (/Users/khaledosman/Documents/github/azure-storage-js-v10-quickstart/index.js:123:85)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

### OS and Version?
> macOS Mojave 10.14.6

### Versions
> node v10.14.1

**Comment by CANHBK** [comment]
i have the same issue

**Comment by mjrousos** [comment]
This happens because the sample calls `.read` on a stream that may not yet be readable. This can be fixed by waiting for `data` or `readable` events on the stream and (only) then reading data. This is demonstrated in [this sample](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/samples/javascript/basic.js#L100) and I fix the issue here as part of PR #5 which updates the sample to use the latest version of `@azure/storage-blob`

**Comment by jongio** [comment]
This should be fixed in the latest version of this sample.  Please reopen if you are able to reproduce.  Thank you
